game/scripts/vscripts/abilities/oaa_kunkka_tidebringer.lua

```diff
@@ -0,0 +1,236 @@
+  return self.BaseClass.GetCooldown( self, level ) - talent_cooldown_reduction
+end
+
+-- Do not allow Rubick to steal this ability
+function kunkka_tidebringer_oaa:IsStealable()
+  return false
+end
+
+function kunkka_tidebringer_oaa:GetIntrinsicModifierName()
+  return "modifier_kunkka_tidebringer_oaa_passive"
+end
+
+--------------------------------------------------------------------------------
+
+function kunkka_tidebringer_oaa:OnUpgrade()
+  -- First point to the ability
+  if self:GetLevel() == 1 then
+    -- Turn on on first level-up
+    if not self:GetAutoCastState() then
+      self:ToggleAutoCast()
+    end
+  end
+end
+
+--------------------------------------------------------------------------------
+
+modifier_kunkka_tidebringer_oaa_passive = class( ModifierBaseClass )
+
+--------------------------------------------------------------------------------
+
+function modifier_kunkka_tidebringer_oaa_passive:IsHidden()
+  return true
+end
+
+function modifier_kunkka_tidebringer_oaa_passive:IsPurgable()
+  return false
+end
+
+function modifier_kunkka_tidebringer_oaa_passive:IsDebuff()
+  return false
+end
+
+function modifier_kunkka_tidebringer_oaa_passive:GetAttributes()
+  return bit.bor(MODIFIER_ATTRIBUTE_IGNORE_INVULNERABLE, MODIFIER_ATTRIBUTE_PERMANENT)
+end
+
+--[[
+function modifier_kunkka_tidebringer_oaa_passive:GetEffectName()
+  return "particles/units/heroes/hero_kunkka/kunkka_weapon_tidebringer.vpcf"
+end
+
+function modifier_kunkka_tidebringer_oaa_passive:GetEffectAttachType()
+  return PATTACH_OVERHEAD_FOLLOW
+end
+
+function modifier_kunkka_tidebringer_oaa_passive:GetStatusEffectName()
+  return "particles/units/heroes/hero_kunkka/kunkka_weapon_tidebringer.vpcf"
+end
+]]
+
+--------------------------------------------------------------------------------
+
+function modifier_kunkka_tidebringer_oaa_passive:OnCreated()
+  local parent = self:GetParent()
+  -- Add weapon glow
+  parent:AddNewModifier(parent, self:GetAbility(), "modifier_kunkka_tidebringer_oaa_weapon_effect", {})
+end
+
+function modifier_kunkka_tidebringer_oaa_passive:DeclareFunctions()
+  return {
+    MODIFIER_EVENT_ON_ATTACK_LANDED,
+    MODIFIER_PROPERTY_PREATTACK_BONUS_DAMAGE
+  }
+end
+
+function modifier_kunkka_tidebringer_oaa_passive:GetModifierPreAttack_BonusDamage( event )
+  if self:GetParent() ~= event.attacker then
+    return
+  end
+
+  -- Toggled on
+  local ability = self:GetAbility()
+  if ability:GetAutoCastState() and ability:IsCooldownReady() then
+    return ability:GetTalentSpecialValueFor("damage_bonus")
+  end
+  return 0
+end
+
+function modifier_kunkka_tidebringer_oaa_passive:OnAttackLanded( event )
+  -- Only attacks FROM parent
+  if self:GetParent() ~= event.attacker then
+    return
+  end
+
+  local parent = self:GetParent()
+  local ability = self:GetAbility()
+
+  -- Toggled off
+  if ( not ability:GetAutoCastState() ) or ( not ability:IsCooldownReady() ) then
```

Comment:
You're missing handling for manual casts of the ability. Refer to Glaives of Wisdom or Arcane Orb for that.

Reply:
I am aware of that.
I still need to verify Magnus's Empower talent which did not work before or Bfury's active.
All 3 of those had lowest priority for me because it works without it.